src/components/Collection.js

```diff
@@ -24,6 +24,12 @@ class Collection extends React.Component {
     this.fetchCollection(nextProps.params.collectionname);
   }
 
+  // this function is a bit hard to read. May be hard to maintain too.
+  // There is at least some repetition below that could factored out.
+  // Break into smaller functions for the different JSX items.
+  // It doesn't look like collectedHeader is being used?
+  // If we make the Button a small React component
+  // maybe we don't need to do bound functions either.
```

Comment:
we agree with this it has been on our todo list for a while now to refactor this code.  Will give it another attempt